File: src/main/java/br/com/blz/testjava/repository/dao/ProductRepository.java
package br.com.blz.testjava.repository.dao;

import java.util.List;

import br.com.blz.testjava.repository.entity.Product;

public interface ProductRepository {

	public Product insert(Product product);
	public Product update(Product product) ;
	public Product findById(long sku);
	public boolean delete(long sku);
	public List<Product> findAll();
	public void removeAll();
	
}

File: src/main/kotlin/com/example/testkotlin/repository/ProductRepository.kt
package com.example.testkotlin.repository

import com.example.testkotlin.model.Product

interface ProductRepository {

    fun findAll():List<Product>
    fun findBySku(sku:Int):Product?
    fun save(p:Product):Boolean
    fun update(p:Product):Boolean
    fun delete(sku:Int):Boolean

}